internal/ls/completions.go

```diff
@@ -1643,6 +1833,57 @@ func keywordCompletionData(
 	}
 }
 
+type resolvingModuleSpecifiersForCompletions struct {
+	logPrefix                      string
+	resolver                       *importSpecifierResolverForCompletions
+	position                       int
+	isForImportStatementCompletion bool
+	isValidTypeOnlyUseSite         bool
+
+	needsFullResolution    bool
+	skippedAny             bool
+	ambientCount           int
+	resolvedCount          int
+	resolvedFromCacheCount int
+	cacheAttemptCount      int
```

Comment:
Can we get rid of all this? We should just make `needsFullResolution` effectively always true and simplify all this. The limiter logic is highly dependent on incomplete completion lists, so that should go away too.